compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp

```diff
@@ -544,9 +545,58 @@ static int32_t getRoundedElementByteWidth(Type type) {
   return llvm::PowerOf2Ceil(byteAligned);
 }
 
-PadEncodingLayoutAttr PadEncodingLayoutAttr::get(MLIRContext *ctx,
-                                                 ArrayRef<int32_t> padding) {
-  return get(ctx, DenseI32ArrayAttr::get(ctx, padding));
+PadEncodingLayoutAttr
+PadEncodingLayoutAttr::get(MLIRContext *ctx, ArrayRef<int32_t> padding,
+                           ArrayRef<ReassociationIndices> reassociation) {
+  // TODO(hanchung): Replace OpBuilder with Builder for upstream
+  // getReassociationIndicesAttribute method.
+  OpBuilder builder(ctx);
+  ArrayAttr reassociationAttr;
+  if (!reassociation.empty() &&
+      llvm::any_of(reassociation, [](auto v) { return v.size() != 1; })) {
+    reassociationAttr =
+        getReassociationIndicesAttribute(builder, reassociation);
+  }
+  auto paddingAttr = builder.getDenseI32ArrayAttr(padding);
+  return Base::get(ctx, paddingAttr, reassociationAttr);
+}
+
+SmallVector<ReassociationIndices>
+PadEncodingLayoutAttr::getReassociationIndices() const {
+  SmallVector<ReassociationIndices> reassociationIndices;
+  auto reassociationAttr = getReassociation();
+
+  // Construct the identity reassociation map if the parameter is not set.
+  if (!reassociationAttr) {
+    for (auto i : llvm::seq<int64_t>(0, getPadding().size())) {
+      reassociationIndices.push_back({i});
+    }
+    return reassociationIndices;
```

Comment:
```suggestion
   llvm::for_each(llvm::seq<int64_t>(0, getPadding().size()), [&](int64_t i) { reassociationIndices.push_back({i}); });
   return rea...
```

Reply:
I don't really enjoy those llvm helpers when the logic is simple and the code can be straightforward. It can be helpful when you have iterators or ranges, but this is just a simple code...

Reply:
There are only 12 uses in IREE: https://github.com/search?q=repo%3Airee-org%2Firee%20for_each&type=code

Unless there is a specific reason, I'm not going to change it.

Reply:
I think we should just adapt to llvm ADT.  I don’t have any strong preference.

Reply:
for_each is not widely used in llvm -- it predates range for loop from c++11